tripper/utils.py

```diff
@@ -84,43 +84,65 @@ def __dir__(self):
         return dict.__dir__(self) + list(self.keys())
 
 
-def recursive_update(d: dict, other: dict, cls: "Optional[type]" = None):
-    """Recursively update dict `d` with dict `other`."""
+def recursive_update(
+    d: dict,
+    other: dict,
+    append: bool = True,
+    cls: "Optional[type]" = None,
+):
+    """Recursively update dict `d` with dict `other`.
+
+    Arguments:
+        d: Dict to update.
+        other: The source to update `d` from.
+        append: Whether to append a keys from `other` to existing keys in `d`.
+            If False, the existing value will be updated.
+        cls: Dict subclass for new sub-dicts in `d`. Defaults to the class
+            of `d`.
+    """
     # pylint: disable=too-many-branches
     if cls is None:
         cls = d.__class__
     if isinstance(other, dict):
         if not isinstance(d, dict):
-            raise TypeError("`d` must be a dict when `other` is a dict")
+            raise TypeError(
```

Comment:
According to the type specification other should always be dicts, and in the docstring d is specified as a dict. Is this not true?